### PR TITLE
update contributing.md

### PR DESCRIPTION
**Pull request by parik2004**
IMPORTANT NOTES (please read and then delete):

* The PR title should start with "Fix #bugnum: " (if applicable), followed by a clear one-line present-tense summary of the changes introduced in the PR. For example: "Fix #bugnum: Changed UI layout from horizontal to vertical.".

* Please make sure to mention "#bugnum" somewhere in the description of the PR. This enables Github to link the PR to the corresponding bug.

Please also make sure to follow the [contribution](https://github.com/Cloud-CV/Fabrik/blob/master/.github/CONTRIBUTING.md) guidelines.

### CODE REVIEW

File: .github/CONTRIBUTING.md [hunk 1]
```diff
@@ -1,26 +1,26 @@
...
-### Setting things up
+### Setting things up:
 
 To set up the development environment, follow the instructions in [README](https://github.com/Cloud-CV/Fabrik/blob/master/README.md).
 
-### Finding something to work on
+### Finding something to work on:
 
-The issue tracker of Fabrik a good place to start. If you find something that interests you, comment on the thread and we’ll help you get started.
+The issue tracker of Fabrik is a good place to start. If you find something that interests you, comment on the thread and we’ll help you to get started.
```
Comment (thatbrguy) [inline .github/CONTRIBUTING.md]:
```suggestion
The issue tracker of Fabrik is a good place to start. If you find something that interests you, comment on the thread and we’ll help you get started.
```
The `to` is not necessary.

File: .github/CONTRIBUTING.md [hunk 2]
```diff
@@ -1,26 +1,26 @@
...
 
 Alternatively, if you come across a new bug in the application, please file a new issue and comment if you would like to be assigned. The existing issues are tagged with one or more labels, based on the part of the application it touches, its importance etc., that can help you in selecting one.
 
-If neither of these seem appealing, please post on our channel and we will help find you something else to work on.
+If neither of these seem appealing, please post on our channel and we'll help you to find something else to work on.
```
Comment (thatbrguy) [inline .github/CONTRIBUTING.md]:
```suggestion
If neither of these seem appealing, please post on our channel and we'll help you find something else to work on.
```
The `to` is not necessary.

File: .github/CONTRIBUTING.md [hunk 3]
```diff
@@ -1,26 +1,26 @@
...
 
-### Instructions to submit code
+### Instructions to submit your code:
 
-Before you submit code, please talk to us via the issue tracker so we know you are working on it.
+Before you submit your code, please talk to us via the issue tracker so that we know you are working on it.
 	
-Our central development branch is master. Coding is done on feature branches based off of development and merged into it once stable and reviewed. To submit code, follow these steps:
+Our central development branch is your master. Coding is done on featured branches based on the development and is then merged into it once stable and reviewed. To submit your code, follow these steps:
```
Comment (thatbrguy) [inline .github/CONTRIBUTING.md]:
```suggestion
Our central development branch is `master`. Coding is done on feature branches based off of development and is then merged into it once stable and reviewed. To submit your code, follow these steps:
```
Minor edits.

File: .github/CONTRIBUTING.md [hunk 4]
```diff
@@ -29,32 +29,32 @@ Our central development branch is master. Coding is done on feature branches bas
...
 	```
 	flake8 ./
 	```
-    - Also, for build errors and checking adherence to `.eslint`, run the following command during development:
+    - Also, for the build errors and to check adherence to `.eslint`, run the following command during development:
 	```
 	webpack --progress --colors --watch
 	```            
-    - If you are using Sublime Text, you can lint code while writing it. Install the following plugins for Sublime Text.
+    - If you are using a Sublime Text, you can lint the code while writing it. Install the following plugins for a Sublime Text.
```
Comment (thatbrguy) [inline .github/CONTRIBUTING.md]:
```suggestion
    - If you are using Sublime Text, you can lint the code while writing it. Install the following plugins for Sublime Text.
```
Minor edit.

File: .github/CONTRIBUTING.md [hunk 5]
```diff
@@ -29,32 +29,32 @@ Our central development branch is master. Coding is done on feature branches bas
...
+    - Commits should be self-contained and they must contain a descriptive commit message.
+    - Please make sure your code is well-formatted and adheres to 'PEP8' conventions (for Python) and the `.eslint` located in the repository (for JavaScript).
     - Please ensure that your code is well tested.
-    - We highly encourage to use `flake8` to follow the PEP8 styling. Run the following command before creating the pull request:
+    - We highly encourage to use `flake8` to follow the 'PEP8' styling. Run the following command before creating the pull request:
 	```
 	flake8 ./
 	```
-    - Also, for build errors and checking adherence to `.eslint`, run the following command during development:
+    - Also, for the build errors and to check adherence to `.eslint`, run the following command during development:
```
Comment (thatbrguy) [inline .github/CONTRIBUTING.md]:
```suggestion
    - Also, for build errors and to check adherence to `.eslint`, run the following command during development:
```
Minor edit.

File: .github/CONTRIBUTING.md [hunk 6]
```diff
@@ -29,32 +29,32 @@ Our central development branch is master. Coding is done on feature branches bas
...
 	```            
-    - If you are using Sublime Text, you can lint code while writing it. Install the following plugins for Sublime Text.
+    - If you are using a Sublime Text, you can lint the code while writing it. Install the following plugins for a Sublime Text.
         ```
         SublimeLinter
         SublimeLinter-contrib-eslint
         SublimeLinter-flake8
         ```
-        *For installing the Sublime Package Control Manager in Sublime-Text Editor use [this](https://packagecontrol.io/installation#st2) link.*
+        *For installing the Sublime Package Control Manager in a Sublime-Text Editor use the link: (https://packagecontrol.io/installation#st2) .*
```
Comment (thatbrguy) [inline .github/CONTRIBUTING.md]:
```suggestion
        *For installing the Sublime Package Control Manager in Sublime-Text Editor, use this [link](https://packagecontrol.io/installation#st2).*
```
Square brackets are used to create hyperlinks in markdown. You can learn more about it [here](https://guides.github.com/features/mastering-markdown/).

File: .github/CONTRIBUTING.md [hunk 7]
```diff
@@ -29,32 +29,32 @@ Our central development branch is master. Coding is done on feature branches bas
...
+    - Check back after some time to see if the Travis checks have passed, if not you should click on “Details” link on your PR thread at the right of “The Travis CI build failed”, which will take you to the dashboard for your PR. You will see what failed or stalled, and you will have to resolve them.
+    - If your checks have passed, your PR will be assigned a reviewer who will review your code and provide comments. Please address each review comment by pushing new commits to the same branch (the PR will automatically update, so you don’t need to submit a new one). Once you are done, comment below each review comment marking it as “Done”. Feel free to use the thread to have a discussion about comments that you don’t understand  or don’t agree with.
 
-    - Once all comments are addressed, the reviewer will give an LGTM (‘looks good to me’) and merge the PR.
+    - Once all comments are addressed, the reviewer will give a LGTM (‘looks good to me’) and merge the PR.
 
-Congratulations, you have successfully contributed to Project Fabrik!
+Congratulations, you have successfully contributed a Project to Fabrik!
```
Comment (thatbrguy) [inline .github/CONTRIBUTING.md]:
```suggestion
Congratulations, you have successfully contributed to Project Fabrik!
```
We refer to Fabrik as Project Fabrik as well :)

**Comment by Ram81** [comment]
No activity in 3 days, closing this for now.

**Pull request closed without merging**